### PR TITLE
Feature/pm 527 cpp client side odrive first setup

### PR DESCRIPTION
**Pull request by Joris-JW**
Closes PM-527

## Description
With this PR it should be possible to control the Odrive using CPP-code and later on ROS control. It is based on the native communication protocol which is also used by the official _odrivetool shell_. This is accomplished by writing a usb driver to send and receive data over the TX/RX communication line of the odrive.

Within this PR a basic program is included to publish data from the Odrive to a ROS topic to generaly display the usage and functionalities of the program. The Json file consists of the default configurations which are set during the startup sequence. 

## Changes
* Include cpp USB driver for the odrive.
* basic program to publish odrive data
* configuration file for the March odrive

### CODE REVIEW

File: odrive_interface/config/odrive_settings/march_odrive.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    {
+        "name":"axis0.config.watchdog_timeout",
+        "type":"float",
+        "value":0
+    },
+    {
+        "name":"axis1.config.watchdog_timeout",
+        "type":"float",
+        "value":0
```
Comment (Roelemans) [inline odrive_interface/config/odrive_settings/march_odrive.json:65]: Hmm 0 meant no watchdog right?

File: odrive_interface/config/odrive_settings/march_odrive.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+[
```
Comment (Roelemans) [inline odrive_interface/config/odrive_settings/march_odrive.json:1]: How do you create this json file? do you have to manually edit it?

File: odrive_interface/include/odrive_interface/odrive.hpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  /**
+   * Execute function on the odrive object
+   */
+  int function(const std::string& function_name);
+
+  /**
+   * Execute function on the odrive object
+   */
+  int setConfigurations(const std::string& configuration_json_path);
```
Comment (Roelemans) [inline odrive_interface/include/odrive_interface/odrive.hpp]: I think you oopsied your docstring here ;)

**Review by Roelemans — CHANGES_REQUESTED** [review]
Looks good! (though I don't 100% understand everything :sweat_smile:).

File: odrive_interface/config/odrive_settings/march_odrive.json (hunk 1 above)
Comment (Joris-JW) [reply odrive_interface/config/odrive_settings/march_odrive.json:65]: yes indeed. When using the odrive u have to manually update the watchdog timer in the hardware interface loop running on the master, eg;
```
  while (ros::ok())
  {
    // update watchdog
    odrives_objects[0].function("axis0.watchdog_feed");
    odrives_objects[1].function("axis1.watchdog_feed");

    r.sleep();
    ros::spinOnce();
  }
```
But i got some problems with the update function so decided to leave it out temporarily.

File: odrive_interface/config/odrive_settings/march_odrive.json (hunk 2 above)
Comment (Joris-JW) [reply odrive_interface/config/odrive_settings/march_odrive.json:1]: I indeed manually created this, seemed like the fasted solution.

File: odrive_interface/include/odrive_interface/odrive.hpp (hunk 3 above)
Comment (Joris-JW) [reply odrive_interface/include/odrive_interface/odrive.hpp]: i did ):

Comment (Roelemans) [reply odrive_interface/include/odrive_interface/odrive.hpp]: How can you be like this

Comment (Roelemans) [reply odrive_interface/include/odrive_interface/odrive.hpp]: ? :'(

**Review by Roelemans — CHANGES_REQUESTED** [review]

File: odrive_interface/include/odrive_interface/odrive.hpp (hunk 3 above)
Comment (Joris-JW) [reply odrive_interface/include/odrive_interface/odrive.hpp]: me so sorry

**Review by Roelemans — APPROVED** [review]

**Pull request merged**